src/fc/fc_monitor.c

```diff
@@ -154,6 +155,41 @@ virFCJsonActionExec(const char *unix_path,
     return response_code;
 }
 
+int virFCMonitorSetConfig(const char *socketpath,
+                          bool hyper_threading,
+                          virDomainDef *vmdef)
+{
+    virJSONValue_autoptr jsonObj = virJSONValueNewObject();
+    long response_code = 0;
+    int ret = -1;
+    int mem_mb = virDomainDefGetMemoryInitial(vmdef);
+    int vcpu_count = 0;
+    int maxvcpus = 0;
+    virDomainVcpuDef *vcpu;
+    size_t i;
+
+    VIR_DEBUG("Memory in mb: %d", mem_mb);
+
+    /* count maximum allowed number vcpus and enabled vcpus when boot.*/
+    maxvcpus = virDomainDefGetVcpusMax(vmdef);
```

Comment:
Firecracker does not support scaling the number of vcpus up or down. Can we enforce this while validating the XML?